meetings/agenda-minutes-2020-04-16.md

```diff
@@ -0,0 +1,76 @@
+- PFC: If you don’t want to expose timezone data the host knows about, then you probably don’t want to reveal the calendars either, so you’d probably patch it to just expose ISO.
+- RGN: It doesn’t feel analogous to adding `Intl.RelativeTimeFormat`. It’s more like adding a locale. It’s less about filling holes and more about patching existing functionality.
+- SFC: Does anyone have other, elegant proposals?
+- DE: Since we also have user-specified time-zones, are we using the same approach there as well?
+- SFC: I actually left a comment on PFC’s pull request about it. We should ideally use a similar approach there.
+- DE: When to convert an ID to calendar, it would create a new calendar. The registry would just return the same objects, which would be different perf-wise.
+- SFC: The registry idea wouldn’t work without frozen calendars.
+- DE: It doesn’t need to be so, not really.
+- RGN: I like the analogy to Intl, and I do like that they’re dealt with similarly. I also like the polyfillability angle. However, I cannot come up with a counter-proposal.
+- DE: What’s the angle on the polyfilability argument?
+- RGN: Adding a single calendar would require a mountain for work, which defeats the purpose.
+- DE: There’s nothing like this in Intl. The subclass-ability, registry approach, etc.
+- SFC, RGN: We’ve not had to deal with those problems.
+- DE: The problem is: I fail to see the use-case here apart from partial implementations?
+- PFC: One of the use-cases for custom time-zones is that if you wanted to do date-time arithmetic before time-zones were standardized.
+- DE: In case of out-of-date implementations, we could redirect the user to the polyfill.
+- SFC: The approach of nuking the existing builtins completely feels awkward to me. You want the ability to patch single calendars or time-zones without getting rid of all the rest.
+- DE: The problem is, there are alternative solutions for all of these things.
+- SFC: That’s why I proposed two things, they’re on either side of the spectrum.
+- DE: I think there’s a cost of all this indirection. My point is, we should have a smaller surface area. The problem is, the stuff around custom calendars and timezones keeps getting more and more complicated and I don’t feel that this is good enough justification.
+- SFC: I now see DE’s point, and I see the connection to Intl. A lot of people built wrappers on top of Intl to deal with the fact that Intl doesn’t have the ability to add data. The difference here is that Temporal is sort of a fundamental data type. Maybe we could move Timezones and Calendars out of Temporal and into a more data-driven namespace.
```

Comment:
> The difference here is that Temporal is sort of a fundamental data type

i disagree "temporal" is treated as fundamental datatype in practice.  subconsciously or not, product-developers usually implement "temporal" as collection of static-functions to achieve some ux-objective (through manipulation and message-passing of serialized isostrings between ui and datastore).

nobody cares about a Date/DateTime-object in and of itself.  the only thing that matters is the isostring-representation  to be integrated with dom-elements, ingested into sql-database, or message-passed between clients and servers.

Reply:
No, they do not usually do that; they usually reach for a library like moment.js which is treated as a fundamental data type.

You are incorrect that the string representation is the only thing that matters; you've asserted this dozens of times and I've attempted to correct your assertion every time, so far with no success.

Reply:
they use moment.js to again, ultimately create static-functions that inputs an isotring, and outputs an isostring (for some ux-specific objective).